include/ucl++.h

```diff
@@ -218,7 +218,7 @@ class Ucl final {
 				cur.reset (new Ucl(ucl_object_iterate_safe (it.get(), true)));
 			}
 
-			if (!*cur) {
```

Comment:
No. The test `!*cur` will call `Ucl::operator bool()` which will return false when the object is a boolean `false`. So iteration over something like this would fail:
```
conf {
    option_a: false
    option_b: 42
}
```
The `cur && ` part may not be needed, as doing anything on an `end`(empty) iterator is undefined behavior anyway.